### PR TITLE
Overhauling billing and usage

### PR DESCRIPTION
**Pull request by crufter**
Everything about this PR is terrible.

Following services must be deployed after the payment service name change here:

```
	modified:   billing/main.go
	modified:   payments/handler/subscriptions.go
	modified:   payments/main.go
	modified:   signup/handler/signup.go
	modified:   subscriptions/main.go
```

### CODE REVIEW

**Comment by asim** [comment]
The other way to handle this.

- Deploy payments service (leave old one where it is)
- Deploy changes to other services that reference payments
- Remove old payment service after everything has been updated
- Zero downtime roll out.

File: .github/workflows/integration-m3o.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@ jobs:
       with:
         repository: 'micro/micro'
         path: 'micro'
+        ref: 'v3.0.0-beta.7'
```
Comment (domwong) [inline .github/workflows/integration-m3o.yml:29]: probably want to remove this

Comment (crufter) [reply .github/workflows/integration-m3o.yml:29]: The problem is latest versions absolutely break the tests with some go vendor thing.
I haven't created a ticket for it but let @ben-toogood know last week when the same happened with `micro/services`

File: billing/handler/billing.go [hunk 2]
```diff
@@ -85,6 +115,12 @@ func NewBilling(ns nsproto.NamespacesService,
 	}
 	maxIncludedServices := val.Int(10)
 
+	val, err = mconfig.Get("micro.billing.report")
+	if err != nil {
+		log.Warnf("Can't load max included services: %v", err)
```
Comment (domwong) [inline billing/handler/billing.go:120]: Seems like the log line doesn't match the error?

File: billing/handler/billing.go [hunk 3]
```diff
@@ -181,6 +211,46 @@ func (b *Billing) Apply(ctx context.Context, req *billing.ApplyRequest, rsp *bil
 	default:
 		return errors.Unauthorized("billing.Apply", "Unauthorized")
 	}
+	if req.All {
+		c := -1
+		for {
+			// prevent infinite loops
+			if c > 100 {
```
Comment (domwong) [inline billing/handler/billing.go:218]: Why do we loop?

Comment (crufter) [reply billing/handler/billing.go:218]: To process all updates. Once we process an update it gets deleted.  Will comment/make it nicer

File: billing/handler/billing.go [hunk 4]
```diff
@@ -244,142 +311,196 @@ func (b *Billing) Portal(ctx context.Context, req *billing.PortalRequest, rsp *b
...
-				planIDToSub := map[string]*sproto.Subscription{}
-				for _, sub := range subsRsp.Subscriptions {
-					planIDToSub[sub.Plan.Id] = sub
-				}
+func deleteUpdate(record *update) error {
+	err := mstore.Delete(fmt.Sprintf("%v%v", updatePrefix, record.ID))
+	if err != nil {
+		return err
+	}
+	if err != nil {
```
Comment (domwong) [inline billing/handler/billing.go:479]: dupe

Comment (crufter) [reply billing/handler/billing.go:479]: eeeee

**Pull request merged**